packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -64,7 +83,7 @@ module.exports = {
     // containing code from all our entry points, and the Webpack runtime.
     filename: 'static/js/bundle.js',
     // In development, we always serve from the root. This makes config easier.
-    publicPath: '/'
+    publicPath: publicPath 
```

Comment:
I can remove this (removes most of the benefit I would get from this pr :( but I understand if it's not suitable here.)

Just want to point out that it's not rails specific really. It's the only way to load your react bundle into a host other than the same host that's serving the js files, Currently we are locked down to webpack serving the html.